platform-api/che-core-api-machine/src/test/java/org/eclipse/che/api/machine/server/recipe/RecipeLoaderTest.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.machine.server.recipe;
+
+import org.eclipse.che.api.core.ServerException;
+import org.eclipse.che.api.machine.server.dao.RecipeDao;
+import org.mockito.Mock;
+import org.mockito.testng.MockitoTestNGListener;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Listeners;
+import org.testng.annotations.Test;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.spy;
+
+/**
+ * @author Anton Korneta
+ */
+@Listeners(MockitoTestNGListener.class)
+public class RecipeLoaderTest {
+
+    private RecipeLoader recipeLoader;
+
+    @Mock
+    private RecipeDao recipeDao;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        recipeLoader = spy(new RecipeLoader("recipes.json", recipeDao));
+    }
+
+    @Test
+    public void shouldLoadPredictableRecipesFromValidJson() throws Exception {
+        doNothing().when(recipeDao).create(any());
+
+        recipeLoader.start();
```

Comment:
It's useful here to check that some required action performed, for example dao.create() is called